app/controllers/admin/customers_controller.rb

```diff
@@ -102,9 +102,11 @@ def customer_tags_by_id
                   context: 'tags' })
 
       customer_tags.each_with_object({}) do |tag, indexed_hash|
-        customer_id = tag.taggings.first.taggable_id
-        indexed_hash[customer_id] ||= []
-        indexed_hash[customer_id] << tag.name
+        tag.taggings.each do |tagging|
```

Comment:
This makes sense to me now. One tag is associated to several customers and we need to look at all of them.

For another PR in the future: It might be easier to load Taggings instead of Tags and iterate over them.